File: LeetCode/findPeakElement.java
/**
 * Created by yiliu on 3/6/15.
 */
public class findPeakElement {

    public int findPeakElement(int[] num) {
        int l = 0;
        int r = num.length - 1;
        int mid = r / 2;
        while (l < r) {
            if (num[mid] < num[mid + 1]) {
                l = mid + 1;
            } else {
                r = mid;
            }
            mid = (l + r) / 2;
        }
        return l;
    }
}

File: LeetCode/Python/FindPeakElement.py
__author__ = 'yi'


class Solution:
    # @param num, a list of integer
    # @return an integer
    def findPeakElement(self, num):
        l = 0
        r = len(num) - 1
        mid = r / 2
        while l < r:
            if num[mid] < num[mid + 1]:
                l = mid + 1
            else:
                r = mid

            mid = (r + l) / 2
        return l
